peepingtom/core/datablocks/base.py

```diff
@@ -1,76 +1,48 @@
 from abc import ABC, abstractmethod
 
+from ...utils.containers import AttributedList
 
-class DataBlock(ABC):
-    """
-    Base class for all simple DataBlock objects, data types which can be visualised by Depictors
-
-    DataBlock objects must implement a data setter method as _data_setter which returns the appropriately formatted data
 
-    Calling __getitem__ on a DataBlock will call __getitem__ on its data property
+class BaseBlock(ABC):
+    """
+    Base class for all simple and complex datablocks.
+    Provides common methods and easy type inference
     """
-    def __init__(self, parent=None):
+    def __init__(self, parent=None, depictor=None):
         self.parent = parent
-
-    @property
-    def data(self):
-        return self._data
-
-    @data.setter
-    def data(self, *args):
-        self._data = self._data_setter(*args)
-        self.updated()
-
-    @abstractmethod
-    def _data_setter(self, data):
-        self.data = data
+        self.depictor = depictor
 
     def dump(self):
         kwargs = {}
-        kwargs.update({'parent': self.parent})
+        kwargs.update({'parent': self.parent, 'depictor': self.depictor})
         return kwargs
 
     def updated(self):
         """
-        this function is called when getitem is used.
-        It is used by other modules to know when the data was changed.
-        When needed, it can be patched with additional callbacks.
+        this function is called when the data changed in order to trigger callbacks
```

Comment:
No. Maybe the docstring should be changed to "this function *should* be called when...", because these callbacks are impplemented only in the daughter class (`DataBlock` for example calls it on `_data_setter`, `setitem` and `delitem`.